bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/AbstractUID.java

```diff
@@ -0,0 +1,129 @@
+package org.eclipse.smarthome.core.common;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * A non specific base class for unique identifiers within the SmartHome framework.
+ *
+ * @author Markus Rathgeb - Splitted from the Thing's UID class
+ */
+@NonNullByDefault
+public abstract class AbstractUID {
+
+    public static final String SEGMENT_PATTERN = "[A-Za-z0-9_-]*";
+    public static final String SEPARATOR = ":";
+    private final List<String> segments;
+
+    /**
+     * Constructor must be public, otherwise it can not be called by subclasses from another package.
+     */
+    public AbstractUID() {
+        this.segments = Collections.emptyList();
+    }
+
+    /**
+     * Parses a UID for a given string. The UID must be in the format
+     * 'bindingId:segment:segment:...'.
+     *
+     * @param uid uid in form a string (must not be null)
+     */
+    public AbstractUID(String uid) {
+        this(splitToSegments(uid));
+    }
+
+    /**
+     * Creates a AbstractUID for a list of segments.
+     *
+     * @param segments the id segments
+     */
+    public AbstractUID(final String... segments) {
+        this(Arrays.asList(segments));
+    }
+
+    /**
+     * Creates a UID for list of segments.
+     *
+     * @param segments segments (must not be null)
+     */
+    public AbstractUID(List<String> segments) {
+        int minNumberOfSegments = getMinimalNumberOfSegments();
+        int numberOfSegments = segments.size();
+        if (numberOfSegments < minNumberOfSegments) {
+            throw new IllegalArgumentException(
+                    String.format("UID must have at least %d segments.", minNumberOfSegments));
+        }
+        for (int i = 0; i < numberOfSegments; i++) {
+            String segment = segments.get(i);
+            validateSegment(segment, i, numberOfSegments);
+        }
+        this.segments = segments;
```

Comment:
It seems I missed one point some time...
The list is supplied by the caller and we should ensure that if he changes the list after the UID generation the internals of the UID object are not changed anymore.
This call should be changed to something similar to: `this.segments = Collections.unmodifiableList(new ArrayList<>(segments));`

Reply:
...as did I! Fixed.